internal/server/deviceflow.go

```diff
@@ -64,77 +64,112 @@ retry:
 }
 
 // GetDeviceFlowStatus is an API handler for checking the status of a device
-// flow login. The response status can be pending, rejected, expired, or confirmed.
+// flow login. The response status can be pending, expired, or confirmed.
 func (a *API) GetDeviceFlowStatus(c *gin.Context, req *api.DeviceFlowStatusRequest) (*api.DeviceFlowStatusResponse, error) {
 	rctx := getRequestContext(c)
-	dfar, err := access.FindDeviceFlowAuthRequest(rctx, req.DeviceCode)
+
+	dfar, err := data.GetDeviceFlowAuthRequest(rctx.DBTxn, data.GetDeviceFlowAuthRequestOptions{ByDeviceCode: req.DeviceCode})
 	if err != nil {
-		return nil, err
+		if errors.Is(err, internal.ErrNotFound) {
+			return nil, err
```

Comment:
This potentially allows someone to scan for device flow requests. Rate limiting should help mitigate that, but a distributed attack could still make lots of guesses. I guess since we delete it once approved it doesn't help them much.

Do we need to differentiate this error, or can we always return unauthorized?

Reply:
Great catch. Let's just return 401 here, as all public endpoints should.